backend/company/serializers.py

```diff
@@ -33,3 +33,14 @@ def create(self, validated_data):
         company.billing_address = billing_address
         company.save()
         return company
+
+    def update(self, company: models.Company, validated_data: dict):
+        company.contact_address.update(validated_data.pop("contact_address"))
+        company.billing_address.update(validated_data.pop("billing_address"))
+        company.update(validated_data)
+        return company
+
+    def validate(self, attrs):
```

Comment:
Phone number validation is done in https://github.com/daviddrysdale/python-phonenumbers/blob/dev/python/phonenumbers/phonenumberutil.py#L2004

Why should API alter data submitted by user?

Reply:
Neviem ci presne rozumiem otazke. Ak ide o phone number tak taky bol poziadavok zakaznika. Ak ide vseobecne o data tak je priordzene ze zakaznik chce aby uzivatel mohol menit udaje o firme.

Reply:
Tuto funkcionalitu by som nechal cisto na frontend, tj ze javascript predvyplni input hodnotou "+420".
API by nemala upravovat data a ukladat ich v inom stave ako boli odoslane

Reply:
Dobre tak to nechame na frontend